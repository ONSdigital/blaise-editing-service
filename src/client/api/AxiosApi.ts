import axios, { AxiosRequestConfig } from 'axios';
import { AuthManager } from 'blaise-login-react-client';
import notFound from '../../server/helpers/axiosHelper';

function axiosConfig(): AxiosRequestConfig {
  const authManager = new AuthManager();
  return {
    headers: {
      'Content-Type': 'application/json',
      ...authManager.authHeader(),
    },
  };
}

export async function getDataFromNode<T>(url: string, notFoundError: string): Promise<T> {
  try {
    const response = await axios.get(url, axiosConfig());

    return response.data;
  } catch (error) {
    if (notFound(error)) {
      throw new Error(notFoundError);
    }
    throw new Error('Unable to complete request, please try again in a few minutes');
  }
}

export async function patchDataToNode(url: string, payload: any, notFoundError: string): Promise<void> {
  try {
    await axios.patch(url, payload, axiosConfig());
  } catch (error) {
    if (notFound(error)) {
      throw new Error(notFoundError);
    }
    throw new Error('Unable to complete request, please try again in a few minutes');
  }
}
