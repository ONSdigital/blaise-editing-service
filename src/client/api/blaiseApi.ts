import axios from 'axios';
import { Questionnaire } from 'blaise-api-node-client';
import { CaseDetails, CaseFactsheet } from '../../common/interfaces/case.interface';
import notFound from '../../common/helpers/axios.helper';

export async function getQuestionnaires(): Promise<Questionnaire[]> {
  try {
    const response = await axios.get('/api/questionnaires');

    return response.data;
  } catch (error) {
    if (notFound(error)) {
      throw new Error('Unable to find questionnaires, please contact Richmond Rice');
    }
    throw new Error('Unable to retrieve questionnaires, please try again in a few minutes');
  }
}

export async function getCases(questionnaireName: string): Promise<CaseDetails[]> {
  try {
    const response = await axios.get(`/api/questionnaires/${questionnaireName}/cases`);
    return response.data;
  } catch (error) {
    if (notFound(error)) {
      throw new Error('The questionnaire is no longer available');
    }
    throw new Error('Unable to retrieve cases, please try again in a few minutes');
  }
}

export async function getCaseFactsheet(questionnaireName: string, caseId: string): Promise<CaseFactsheet> {
  try {
    const response = await axios.get(`/api/questionnaires/${questionnaireName}/cases/${caseId}/factsheet`);
    return response.data;
  } catch (error) {
    if (notFound(error)) {
      throw new Error('The questionnaire is no longer available');
    }
    throw new Error('Unable to retrieve case factsheet, please try again in a few minutes');
  }
}
