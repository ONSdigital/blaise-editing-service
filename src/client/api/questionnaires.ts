import axios from 'axios';
import { GetQuestionnaires } from '../../shared/responses/GetQuestionnaires';

export default async function getQuestionnaires(): Promise<GetQuestionnaires> {
  const response = await axios.get('/api/questionnaires');
  return response.data;
}
