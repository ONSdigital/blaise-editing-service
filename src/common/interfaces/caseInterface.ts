export interface CaseSummaryDetails {
  CaseId: string
  OutcomeCode: number,
  InterviewDate: Date,
  InterviewerName: string,
  InterviewerNumber: string,
  NumberOfRespondents: number,
  Household: {
    Address: {
      AddressLine1: string,
      AddressLine2: string,
      AddressLine3: string,
      AddressLine4: string,
      County: string,
      Town: string,
      Postcode: string,
    },
    Type: string,
    FloorNumber: number,
    Status: string,
    NumberOfBedrooms: number,
    ReceiptOfHousingBenefit: number,
    PeriodCode: number,
    CouncilTaxBand: string,
    BusinessRoom: boolean,
    SelfEmployed: boolean,
    SelfEmployedMembers: string,
    IncomeSupport: boolean,
    IncomeSupportMembers: string,
    IncomeBasesJaSupport: boolean,
    IncomeBasesJaSupportMembers: string,

  },
  Respondents: {
    RespondentName: string,
    DateOfBirth: Date,
  }[]
}
