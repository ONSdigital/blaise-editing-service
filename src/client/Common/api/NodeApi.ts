import axios from 'axios';
import notFound from '../../../server/helpers/axiosHelper';
import { Survey } from '../../../common/interfaces/surveyInterface';

async function getDataFromNode<T>(url: string, notFoundError: string): Promise<T> {
  try {
    const response = await axios.get(url);

    return response.data;
  } catch (error) {
    if (notFound(error)) {
      throw new Error(notFoundError);
    }
    throw new Error('Unable to complete request, please try again in a few minutes');
  }
}

export default async function getSurveys(): Promise<Survey[]> {
  return getDataFromNode('/api/surveys', 'Unable to find surveys, please contact Richmond Rice');
}
