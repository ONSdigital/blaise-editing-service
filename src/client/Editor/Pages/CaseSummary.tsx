import { ReactElement } from 'react';
import { useParams } from 'react-router-dom';
// import { getCaseSummary } from '../../api/NodeApi';
// import { useAsyncRequestWithTwoParams } from '../../Common/hooks/useAsyncRequest';
// import AsyncContent from '../../Common/components/AsyncContent';
// import { CaseSummaryDetails } from '../../../common/interfaces/caseInterface';
import CaseSummaryContent from '../Components/CaseSummaryContent';
import { CaseSummaryDetails } from '../../../common/interfaces/caseInterface';

function DisplayCaseSummary(questionnaireName: string, caseId: string) {
  // const caseSummary = useAsyncRequestWithTwoParams<CaseSummaryDetails, string, string>(getCaseSummary, questionnaireName, caseId);
  console.log(questionnaireName, caseId);
  const caseSummaryDetailsMockObject: CaseSummaryDetails = {
    CaseId: '90001',
    OutcomeCode: 110,
    InterviewDate: new Date(2024, 5, 11),
    District: 'Gwent',
    InterviewerName: 'Rich',
    NumberOfRespondents: 2,
    Household: {
      Type: '',
      FloorNumber: 0,
      Status: '',
      NumberOfBedrooms: 1,
      ReceiptOfHousingBenefit: 380,
      PeriodCode: 380,
      CouncilTaxBand: 'band A',
      BusinessRoom: false,
      SelfEmployed: false,
      SelfEmployedMembers: '',
      IncomeSupport: false,
      IncomeSupportMembers: '',
      IncomeBasesJaSupport: false,
      IncomeBasesJaSupportMembers: '',
    },
    Respondents: [
      {
        RespondentName: 'Richmond Ricecake',
        DateOfBirth: new Date(1980, 1, 15),
      },
      {
        RespondentName: 'Bartholomew Edgar',
        DateOfBirth: new Date(1995, 5, 11),
      },
    ],
  };

  return (
    <div data-testid="Summary">
      {/*       <AsyncContent content={caseSummary}>
        {(caseSummaryContent) => <CaseSummaryContent caseSummary={caseSummaryContent} />}
      </AsyncContent> */}

      <CaseSummaryContent caseSummary={caseSummaryDetailsMockObject} />
    </div>
  );
}

export type CaseSummaryParams = {
  questionnaireName: string
  caseId: string
};

export default function CaseSummary(): ReactElement {
  const { questionnaireName, caseId } = useParams<keyof CaseSummaryParams>() as CaseSummaryParams;

  return DisplayCaseSummary(questionnaireName, caseId);
}
