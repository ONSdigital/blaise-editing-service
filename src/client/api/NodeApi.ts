import axios from 'axios';

import { CaseEditInformation } from 'blaise-api-node-client/lib/cjs/interfaces/case';
import { User } from 'blaise-api-node-client/lib/cjs/interfaces/user';
import notFound from '../../server/helpers/axiosHelper';
import { Survey } from '../../common/interfaces/surveyInterface';
import { SupervisorInformation } from '../Interfaces/supervisorInterface';
import { EditorInformation } from '../Interfaces/editorInterface';
import mapEditorInformation from '../Mappers/editorInformaitionMapper';
import mapSupervisorInformation from '../Mappers/supervisorInformationMapper';
import { CaseSummaryDetails } from '../../common/interfaces/caseInterface';
import { CasesNotAllocatedInformation } from '../Interfaces/caseAllocationInterface';
import mapCasesNotAllocated from '../Mappers/caseAllocationMapper';
// import { caseSummaryDetailsMockObject } from '../../test/server/mockObjects/CaseMockObject';

async function getDataFromNode<T>(url: string, notFoundError: string): Promise<T> {
  try {
    const response = await axios.get(url);

    return response.data;
  } catch (error) {
    if (notFound(error)) {
      throw new Error(notFoundError);
    }
    throw new Error('Unable to complete request, please try again in a few minutes');
  }
}

export async function getSurveys(userRole: string): Promise<Survey[]> {
  return getDataFromNode(`/api/surveys?userRole=${userRole}`, 'Unable to find surveys, please contact Richmond Rice');
}

export async function getCaseSummary(questionnaireName: string, caseId: string): Promise<CaseSummaryDetails> {
  return getDataFromNode(`/api/questionnaires/${questionnaireName.toUpperCase()}/cases/${caseId}/summary`, 'The questionnaire is no longer available');
}

async function getCaseEditInformation(questionnaireName: string, userRole: string) {
  // TODO Fix the URL upper
  return getDataFromNode<CaseEditInformation[]>(`/api/questionnaires/${questionnaireName.toUpperCase()}/cases/edit?userRole=${userRole}`, 'Unable to find case edit information, please contact Richmond Rice');
}

export async function getEditorInformation(questionnaireName: string, editorUsername:string, editorRole: string): Promise<EditorInformation> {
  const caseEditInformationList = await getCaseEditInformation(questionnaireName, editorRole);
  const caseEditInformationListForEditor = caseEditInformationList.filter((caseEditInformation) => caseEditInformation.assignedTo === editorUsername);

  return mapEditorInformation(caseEditInformationListForEditor);
}

export async function getSupervisorEditorInformation(questionnaireName: string, supervisorRole: string, editorRole: string): Promise<SupervisorInformation> {
  const caseEditInformationList = await getCaseEditInformation(questionnaireName, supervisorRole);
  const editors = await getDataFromNode<User[]>(`/api/users?userRole=${editorRole}`, 'Unable to find user information, please contact Richmond Rice');

  return mapSupervisorInformation(caseEditInformationList, editors);
}

export async function getCasesNotAllocatedInformation(questionnaireName: string, supervisorRole: string, editorRole: string): Promise<CasesNotAllocatedInformation> {
  const caseEditInformationList = await getCaseEditInformation(questionnaireName, supervisorRole);
  const editors = await getDataFromNode<User[]>(`/api/users?userRole=${editorRole}`, 'Unable to find user information, please contact Richmond Rice');

  return mapCasesNotAllocated(caseEditInformationList, editors);
}
