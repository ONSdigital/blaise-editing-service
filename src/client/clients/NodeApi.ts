import axios from 'axios';
import { CaseDetails, CaseFactsheetDetails } from '../../common/interfaces/caseInterface';
import notFound from '../../common/helpers/axiosHelper';
import { Survey } from '../../common/interfaces/surveyInterface';

/* eslint-disable class-methods-use-this */
export default class NodeApi {
  constructor() {
    this.getSurveys = this.getSurveys.bind(this);
    this.getCases = this.getCases.bind(this);
    this.getCaseFactsheet = this.getCaseFactsheet.bind(this);
  }

  async getDataFromNode<T>(url: string, notFoundError: string): Promise<T> {
    try {
      const response = await axios.get(url);

      return response.data;
    } catch (error) {
      if (notFound(error)) {
        throw new Error(notFoundError);
      }
      throw new Error('Unable to complete request, please try again in a few minutes');
    }
  }

  async getSurveys(): Promise<Survey[]> {
    return this.getDataFromNode('/api/surveys', 'Unable to find surveys, please contact Richmond Rice');
  }

  async getCases(questionnaireName: string): Promise<CaseDetails[]> {
    return this.getDataFromNode(`/api/questionnaires/${questionnaireName}/cases`, 'The questionnaire is no longer available');
  }

  async getCaseFactsheet(questionnaireName: string, caseId: string): Promise<CaseFactsheetDetails> {
    return this.getDataFromNode(`/api/questionnaires/${questionnaireName}/cases/${caseId}/factsheet`, 'The questionnaire is no longer available');
  }
}
