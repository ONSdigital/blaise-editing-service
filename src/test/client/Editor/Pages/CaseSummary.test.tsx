import { RenderResult, act, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Router from 'react-router';
import { CaseSummaryDetails } from '../../../../common/interfaces/caseInterface';
import CaseSummary from '../../../../client/Editor/Pages/CaseSummary';
import { getCaseSummary } from '../../../../client/api/NodeApi';
import { caseSummaryDetailsMockObject } from '../../../server/mockObjects/CaseMockObject';

// declare global vars
const questionnaireName: string = 'TEST111A';
const caseId: string = '1';
let view:RenderResult;

// declare mocks
/* eslint import/no-extraneous-dependencies: 0 */
jest.mock('react-router', () => ({ ...jest.requireActual('react-router'), useParams: jest.fn() }));
jest.spyOn(Router, 'useParams').mockReturnValue({ questionnaireName, caseId });
jest.mock('../../../../client/api/NodeApi');
const getCaseSummaryDetailsMock = getCaseSummary as jest.Mock<Promise<CaseSummaryDetails>>;

describe('Given there is a case available in blaise for a questionnaire', () => {
  afterEach(() => {
    getCaseSummaryDetailsMock.mockReset();
  });

  it('should render the summary page for the case correctly', async () => {
    // arrange
    const expectedCaseSummaryDetails: CaseSummaryDetails = caseSummaryDetailsMockObject;

    getCaseSummaryDetailsMock.mockImplementation(() => Promise.resolve(expectedCaseSummaryDetails));

    // act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <CaseSummary />
        </BrowserRouter>,
      );
    });

    // assert
    const caseSummaryView = view.getByTestId('Summary');
    expect(caseSummaryView).toHaveTextContent(expectedCaseSummaryDetails.CaseId);
    expect(caseSummaryView).toHaveTextContent(expectedCaseSummaryDetails.Address.AddressLine1);
    expect(caseSummaryView).toHaveTextContent(expectedCaseSummaryDetails.Address.AddressLine2);
    expect(caseSummaryView).toHaveTextContent(expectedCaseSummaryDetails.Address.AddressLine3);
    expect(caseSummaryView).toHaveTextContent(expectedCaseSummaryDetails.Address.AddressLine4);
    expect(caseSummaryView).toHaveTextContent(expectedCaseSummaryDetails.Address.County);
    expect(caseSummaryView).toHaveTextContent(expectedCaseSummaryDetails.Address.Postcode);
    expect(caseSummaryView).toHaveTextContent(expectedCaseSummaryDetails.Address.Town);
    expect(caseSummaryView).toHaveTextContent(expectedCaseSummaryDetails.InterviewerName);
    expect(caseSummaryView).toHaveTextContent(String(expectedCaseSummaryDetails.NumberOfRespondents));
    expect(caseSummaryView).toHaveTextContent(String(expectedCaseSummaryDetails.OutcomeCode));
    expectedCaseSummaryDetails.Respondents.forEach((respondent) => {
      expect(caseSummaryView).toHaveTextContent(respondent.RespondentName);
      expect(caseSummaryView).toHaveTextContent(respondent.DateOfBirth.toDateString());
    });
  });

  it('should display the summary correctly', async () => {
    // arrange
    const expectedCaseSummaryDetails: CaseSummaryDetails = caseSummaryDetailsMockObject;

    getCaseSummaryDetailsMock.mockImplementation(() => Promise.resolve(expectedCaseSummaryDetails));

    // act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <CaseSummary />
        </BrowserRouter>,
      );
    });

    // assert
    expect(view).toMatchSnapshot();
  });
});

/* describe('Given there the blaise rest api is not available', () => {
  beforeEach(() => {
    getCaseSummaryDetailsMock.mockRejectedValue(new Error('try again in a few minutes'));
  });

  afterEach(() => {
    getCaseSummaryDetailsMock.mockReset();
  });

  it('should display an error message telling the user to try again in a few minutes', async () => {
    // act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <CaseSummary />
        </BrowserRouter>,
      );
    });

    // assert
    const summaryView = view.getByTestId('Summary');
    expect(summaryView).toHaveTextContent('try again in a few minutes');
  });

  it('should render the page correctly when an error occurs', async () => {
    // act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <CaseSummary />
        </BrowserRouter>,
      );
    });

    // assert
    expect(view).toMatchSnapshot();
  });
});
 */