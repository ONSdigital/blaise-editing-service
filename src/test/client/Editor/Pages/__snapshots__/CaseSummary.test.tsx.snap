// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given there is a case available in blaise for a questionnaire should display the summary correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="Summary"
      >
        <div
          class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
        >
          <div
            class="ons-panel__body "
          >
            Case summary for
             
            <strong>
              90001
            </strong>
             
            completed by
             
            <strong>
              Rich
            </strong>
          </div>
        </div>
        <dl
          class="ons-metadata ons-metadata__list ons-grid ons-grid--gutterless ons-u-cf ons-u-mb-no"
          style="margin: 2% 5% 2% 5%;"
          title="summary"
        >
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Serial Number
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            90001
          </dd>
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Outcome code
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            110
          </dd>
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Interview date
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Tue Jun 11 2024
          </dd>
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            District
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Gwent
          </dd>
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Interviewer number
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            2100
          </dd>
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Interviewer name
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Rich
          </dd>
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Address:
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Flat 1
          </dd>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Richmond House
          </dd>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Rice Road
          </dd>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Duffrin
          </dd>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Newport
          </dd>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Gwent
          </dd>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            NZ11 4PD
          </dd>
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Type of accommodation
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          />
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Floor number
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            0
          </dd>
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Household status
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          />
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Number of bedrooms
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            1
          </dd>
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Receipt of housing benefit
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            380
          </dd>
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Period code
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            380
          </dd>
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Council tax band
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            band A
          </dd>
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Business room?
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          />
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Anyone self emplyed?
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          />
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            H/H members
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          />
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Income support received now
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          />
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            H/H members
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          />
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Income based JA received now
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          />
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            H/H members
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          />
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Household size:
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            2
          </dd>
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Respondents:
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Richmond Ricecake
             
            Fri Feb 15 1980
          </dd>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Bartholomew Edgar
             
            Sun Jun 11 1995
          </dd>
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Notes:
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            These are the notes that the editor has made and has been extracted from the quetsionnarie. They may be able to be edited and saved again in this screen. If this is what you wish to do?
          </dd>
        </dl>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="Summary"
    >
      <div
        class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
      >
        <div
          class="ons-panel__body "
        >
          Case summary for
           
          <strong>
            90001
          </strong>
           
          completed by
           
          <strong>
            Rich
          </strong>
        </div>
      </div>
      <dl
        class="ons-metadata ons-metadata__list ons-grid ons-grid--gutterless ons-u-cf ons-u-mb-no"
        style="margin: 2% 5% 2% 5%;"
        title="summary"
      >
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Serial Number
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          90001
        </dd>
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Outcome code
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          110
        </dd>
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Interview date
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Tue Jun 11 2024
        </dd>
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          District
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Gwent
        </dd>
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Interviewer number
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          2100
        </dd>
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Interviewer name
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Rich
        </dd>
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Address:
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Flat 1
        </dd>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Richmond House
        </dd>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Rice Road
        </dd>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Duffrin
        </dd>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Newport
        </dd>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Gwent
        </dd>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          NZ11 4PD
        </dd>
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Type of accommodation
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        />
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Floor number
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          0
        </dd>
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Household status
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        />
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Number of bedrooms
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          1
        </dd>
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Receipt of housing benefit
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          380
        </dd>
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Period code
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          380
        </dd>
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Council tax band
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          band A
        </dd>
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Business room?
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        />
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Anyone self emplyed?
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        />
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          H/H members
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        />
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Income support received now
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        />
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          H/H members
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        />
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Income based JA received now
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        />
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          H/H members
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        />
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Household size:
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          2
        </dd>
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Respondents:
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Richmond Ricecake
           
          Fri Feb 15 1980
        </dd>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Bartholomew Edgar
           
          Sun Jun 11 1995
        </dd>
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Notes:
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          These are the notes that the editor has made and has been extracted from the quetsionnarie. They may be able to be edited and saved again in this screen. If this is what you wish to do?
        </dd>
      </dl>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
